src/Functions/parseDateTime.cpp

```diff
@@ -105,40 +105,65 @@ namespace
 
         std::variant<Int32, Int64, Pos> res;
 
+        /// Helper to disambiguate the first Status argument: Pos, Int64, Int32 (for success) vs. int (for errors)
+        struct IsValueTag {};
+
         Status() = default;
-        Status(Pos pos_) : res(pos_) { }
-        Status(Int64 pos_) : res(pos_) { }
-        Status(Int32 pos_) : res(pos_) { }
+        Status(IsValueTag /*tag*/, Pos pos_) : res(pos_) { }
```

Comment:
Good method. I learned it. I have been struggling with how to distinguish error code and result value.